### PR TITLE
Fix memory error in psi::MultipolePropCalc::compute_mo_extents

### PR DESCRIPTION
**Pull request by raimis**
## Description
This is part of *Psi4* porting to Windows (#933).

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] Repeat `ci-property` test to catch random failures
- [x] Remove memory error in `psi::MultipolePropCalc::compute_mo_extents`.
- [x] Fix `ci-property` test on Windows

## Questions
- [x] Removed some dead code, including the part with `C_DDOT`, which was segment-faulting. Could you check, it was really a dead code, not some incomplete implementation? --> Dead!

## Checklist
- [x] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/libmints/oeprop.cc [hunk 1]
```diff
@@ -1432,17 +1432,8 @@ void OEProp::compute_mo_extents() {
...
-        Ca = Ca_ao();
-        Cb = Ca;
-    } else {
-        Ca = Ca_ao();
-        Cb = Cb_ao();
-    }
+    SharedMatrix Ca = Ca_ao();
 
     std::vector<SharedVector> mo_es;
     mo_es.push_back(SharedVector(new Vector("<x^2>", basisset_->nbf())));
```
Comment (robertodr) [inline psi4/src/psi4/libmints/oeprop.cc]: Could you convert to `std::make_shared<Vector>("<x^2>", basisset_->nbf())`? I am actually surprised the construct using `new` is still around...

File: psi4/src/psi4/libmints/oeprop.cc [hunk 2]
```diff
@@ -1466,51 +1451,23 @@ std::vector<SharedVector> MultipolePropCalc::compute_mo_extents(bool print_outpu
...
-        for (int i = 0; i < nmo; i++) {
-            dipole[1]->set(0, i, C_DDOT(nao, Ca->pointer()[i], nmo, temp->pointer()[i], 1));
-        }
-        C_DGEMM('T', 'N', nmo, nao, nao, 1.0, Ca->pointer()[0], nmo, ao_Dpole[2]->pointer()[0], nao, 0.0,
-                temp->pointer()[0], nao);
-        for (int i = 0; i < nmo; i++) {
-            dipole[2]->set(0, i, C_DDOT(nao, Ca->pointer()[i], nmo, temp->pointer()[i], 1));
-        }
 // Quadrupoles
 #if 0
```
Comment (robertodr) [inline psi4/src/psi4/libmints/oeprop.cc]: This preprocessor `if`/`else` can be dropped, I think. Only the branch after the `else` is actually used.

File: psi4/src/psi4/libmints/oeprop.cc [hunk 3]
```diff
@@ -1466,51 +1451,23 @@ std::vector<SharedVector> MultipolePropCalc::compute_mo_extents(bool print_outpu
...
-
     std::vector<SharedVector> quadrupole;
     quadrupole.push_back(std::make_shared<Vector>("Orbital Quadrupole XX", Ca->ncol()));
     quadrupole.push_back(std::make_shared<Vector>("Orbital Quadrupole YY", Ca->ncol()));
     quadrupole.push_back(std::make_shared<Vector>("Orbital Quadrupole ZZ", Ca->ncol()));
 
-    auto temp = std::make_shared<Matrix>("Temp", Ca->nrow(), Ca->ncol());
-
-    int nao = Ca->nrow();
+    // auto temp = std::make_shared<Matrix>("Temp", Ca->nrow(), Ca->ncol());
```
Comment (robertodr) [inline psi4/src/psi4/libmints/oeprop.cc]: Just remove these two lines.

**Comment by loriab** [comment]
These two commits (search `mo_ext`; Rob https://github.com/psi4/psi4archive/commit/a6153df40ec424e5fdc75fb037b8a3c3fd618484 and Jet https://github.com/psi4/psi4archive/commit/d337912cc48fe14aa8ae78820581fc8f0495383a) look to be the main ones in the mo_extents history. (Also Lori https://github.com/psi4/psi4archive/commit/79e09e2ed1092437bc80cabaf13acaad12686a49 from a psiworkweek but that wasn't operational.)

From comment `// TODO: Print polarity <\vec x> and extents <\vec x^2> - <\vec x>^2` and [earlier print statements](https://github.com/psi4/psi4archive/commit/79e09e2ed1092437bc80cabaf13acaad12686a49#diff-2d2260d0be9213bdd2d346178da27329R518), it sounds like the dipole part was meant for polarity printing. So it should be safe to remove or move to another function rather than tagging along in the extents function.

File: psi4/src/psi4/libmints/oeprop.cc (hunk 1 above)
Comment (raimis) [reply psi4/src/psi4/libmints/oeprop.cc]: Done!

**Comment by raimis** [comment]
This is ready to be merged!

**Pull request merged**